tools/openocd/defs.bzl

```diff
@@ -92,20 +94,66 @@ def _openocd_debug_server_impl(ctx):
     if ctx.attr.transport:
         transport_string = "-c \"transport select %s\"" % (ctx.attr.transport)
     script_template = """
-{openocd} {interface_config_string} {transport_string} -c "adapter_khz {programmer_frequency}" {chip_config_string}
+{openocd} {interface_config_string} {transport_string} -c "adapter speed {programmer_frequency}" {chip_config_string}
 
 """
 
-    script = ctx.actions.declare_file("%s.sh" % ctx.label.name)
-    script_content = script_template.format(
-        openocd = ctx.file._openocd.short_path,
-        interface_config_string = interface_config_string,
-        chip_config_string = chip_config_string,
-        programmer_frequency = ctx.attr.programmer_frequency,
-        transport_string = transport_string,
-    )
+    script_content = ""
+    if is_windows:
+        # A bit of a hacky way to do this... The issue is Windows does not support
+        # symlinks which does not work well for runfiles in particular. To enable them
+        # the user must use Developer Mode, also not ideal. Instead:
+        # 1) Use the 'symlink' as a generic path to the root of openocd
+        # 2) Open the MANIFEST file (non-standard format, beware for future versions)
+        # 3) Find com_openocd and use the full path to the symlink
+        # 4) In batch script, cd to base of tool and call bin/openocd.exe directly
+        windows_script_prepend = """
```

Comment:
I'm going to have to properly learn batch scripting before I can reasonably review this. I don't fully understand what is going on here.

Reply:
I'm added more comments to this snippet of code, and I'll move it out to a common function. I need the comments just to remember exactly how it works and I just put it together yesterday. Maybe someone more familiar with batch scripts can make sure I'm not totally off base :)

Reply:
No worries. I've got a bit more time over the next couple days, so I should have a bit more time for reviews.